src/com/tapia/foss/frames/Visualizer.java

```diff
@@ -232,7 +238,13 @@ private void jButtonAddActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FI
         }
         else
         {
-            structure.add(Integer.parseInt(input));
+            
+            if (this.tryParsing(input)) {
+                structure.add(Integer.parseInt(input));
+            } else {
+                return;
```

Comment:
The return is not necessary unless calling display() after a parsing failure is a problem, but I don't think it is.  Will replace with the JOptionPane notification that was in the "tryParsing" method before.